WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderTrackerInfo.kt

```diff
@@ -0,0 +1,34 @@
+package org.wordpress.android.ui.reader
+
+import org.wordpress.android.ui.reader.utils.DateProvider
+import org.wordpress.android.util.DateTimeUtils
+import java.util.Date
+
+sealed class ReaderTrackerInfo(val key: String, var accumulatedTime: Int, private val dateProvider: DateProvider) {
+    private var startTime: Date? = null
+
+    fun init() {
+        accumulatedTime = 0
+        startTime = null
+    }
+
+    fun start() {
+        startTime = dateProvider.getCurrentTime()
+    }
+
+    fun stop() {
+        startTime?.let {
```

Comment:
In theory AFAIU it should not be possible to get the null if the functions sequence start->stop is always preserved. My reason to keep the startTime to null was for:
- ensure the start/stop sequence was preserved; but I didn't want to fast fail for a tracking tbh
- don't want to false count something has not been started first, but to log this "unexpected case" makes sense. Added log here d7f9c2d2f36c0c2d1c7375a99efac8a4aab2eebb

Reply:
One thing I forgot to report is that it would be good to me to have a log in Sentry (not a crash but only an event); I was looking into the `CrashLoggingUtils.log` but was not fully sure it was what we need, maybe you have some suggestion on this? Thanks in advance 😊.

Reply:
>ensure the start/stop sequence was preserved; but I didn't want to fast fail for this tracking tbh, to avoid crashing for some possible edge case I could possibly not have thought to and not related to the caller logic not keeping the correct/expected sequence

What I like to do in these cases is to wrap the crash/exception in `BuildType.Debug` so it crashes only in a debug build and log an exception in the release build. Having said that I think keeping just the log for this scenario is completely fine;).

>I was looking into the CrashLoggingUtils.log

I think using both `CrashLoggingUtils.log` or `CrashLoggingUtils.logException` is fine. They send an error event to Sentry but it's marked as "Error" not "Fatal" -> it means we can see in Sentry that it's not a crash, but just a log.